lib/ansible/plugins/connection/docker.py

```diff
@@ -132,8 +132,8 @@ def _get_docker_version(self):
 
         cmd, cmd_output, err, returncode = self._old_docker_version()
         if returncode == 0:
-            for line in cmd_output.split('\n'):
-                if line.startswith('Server version:'):  # old docker versions
+            for line in cmd_output.split(b'\n'):
+                if line.startswith(b'Server version:'):  # old docker versions
```

Comment:
Ah, I see, but this is an internal method, not part of the API. Hmm...  After checking the code that uses this function, you could do this with to_text() or to_native() here instead of dealing in byte strings.  That probably would be better.

Do you also have to perform the same conversion after line 139 (where we call self._new_docker_version)?